hack/ci/run-konnectivity-e2e-test.sh

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The Kubermatic Kubernetes Platform contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+### This script is used as a postsubmit job and updates the dev master
+### cluster after every commit to master.
+
+set -euo pipefail
+
+cd $(dirname $0)/../..
+source hack/lib.sh
+
+function cleanup() {
+  if [[ -n "${TMP:-}" ]]; then
+    rm -rf "${TMP}"
+  fi
+}
+trap cleanup EXIT SIGINT SIGTERM
+
+export KUBERMATIC_DEX_VALUES_FILE=${KUBERMATIC_DEX_VALUES_FILE:-$(realpath hack/ci/testdata/konnectivity_oauth_values.yaml)}
+export KUBERMATIC_API_ENDPOINT=${KUBERMATIC_API_ENDPOINT:-https://dev.kubermatic.io}
+
+export GIT_HEAD_HASH="$(git rev-parse HEAD | tr -d '\n')"
+
+echodate "Getting secrets from Vault"
+retry 5 vault_ci_login
+
+if [ -z "${KUBECONFIG:-}" ]; then
+  echodate "No \$KUBECONFIG set, defaulting to Vault key dev/seed-clusters/dev.kubermatic.io"
+
+  KUBECONFIG="$(mktemp)"
+  vault kv get -field=kubeconfig dev/seed-clusters/dev.kubermatic.io > $KUBECONFIG
+fi
```

Comment:
So this actually runs the test on the dev cluster? Is that something that we can / should do in CI?
I see that other e2e tests usually [run in kind](https://github.com/kubermatic/kubermatic/blob/69d12dbb03d6524f150be438f9c9030669d0f835/hack/ci/run-opa-e2e-tests.sh#L34-L35) which makes sense for CI - to not depend on our dev environment.

Reply:
> So this actually runs the test on the dev cluster? 

It runs as a cron job. Instead of kind it uses the dev cluster. 

> Is that something that we can / should do in CI?

We can with kind. 

Should we? Probably not. See discussion https://kubermatic.slack.com/archives/C01EWQZEW69/p1641891094003200 

>I see that other e2e tests usually run in kind which makes sense for CI - to not depend on our dev environment.

Right. If we ran it in CI then definitely doesn't make sense to involve the dev cluster.

Reply:
Other periodic jobs don't use dev, this one shouldn't either IMHO. dev might be in a different state than KKP `master` because someone is testing stuff manually. The setup scripts for KKP in kind should make this relatively straightforward.

Reply:
Plus, you might wanna make this e2e test available for running on a PR, so changes to konnectivity can be e2e tested before being merged.